alf/config_util.py

```diff
@@ -596,6 +606,73 @@ def _decorate(fn_or_cls, name, whitelist, blacklist):
         return fn_or_cls
 
 
+def repr_wrapper(cls):
+    """A wrapper for automatically generating readable repr for an object.
```

Comment:
It's better to clarify that the repr shows how the object is created with non-default argument values (but not other information like class members)